src/transaction/components/TransactionProvider.tsx

```diff
@@ -62,13 +68,33 @@ export function TransactionProvider({
   }); // Component lifecycle
   const [transactionId, setTransactionId] = useState('');
   const [transactionHashList, setTransactionHashList] = useState<Address[]>([]);
+  const [transactions, setTransactions] = useState<
+    Call[] | ContractFunctionParameters[] | undefined
+  >([]);
 
   // Retrieve wallet capabilities
   const { walletCapabilities } = useOnchainKit();
 
   const { switchChainAsync } = useSwitchChain();
 
-  // Hooks that depend from Core Hooks
+  // Validate `calls` and `contracts` props
+  if (!contracts && !calls) {
+    throw new Error('Transaction: One of contracts or calls must be provided.');
```

Comment:
Let's try to make those two sentence even more clear.

Something like `in the component X bla bla in prop bla bla`, like literally food spoon the answear.

Reply:
should we also enforce this with typescript?